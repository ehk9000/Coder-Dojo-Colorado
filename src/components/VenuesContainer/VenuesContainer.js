import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import VenueSearchBar from '../VenueSearchBar/VenueSearchBar';

export function VenuesContainer({ venues, isLoading, error }) {
  const generatedVenues = () => venues
    .map(venue => (
      <article key={venue.id}>
        <h3>
          {venue.name}
        </h3>
        <p>{venue.notes}</p>
      </article> 
    ));
  
  return (
    <section>
      <VenueSearchBar />
      { isLoading && <h3>Loading dojos...</h3>}
      { error && <h4>{error}</h4>}
      { generatedVenues() }
    </section>
  );
}

export const mapStateToProps = state => ({
  venues: state.searchResults,
  isLoading: state.isFetching,
  error: state.error
});

VenuesContainer.propTypes = {
  venues: PropTypes.array,
  isLoading: PropTypes.bool,
  error: PropTypes.string
};

VenuesContainer.defaultProps = {
  venues: [],
  isLoading: false,
  error: ''
};

export default connect(mapStateToProps)(VenuesContainer);
