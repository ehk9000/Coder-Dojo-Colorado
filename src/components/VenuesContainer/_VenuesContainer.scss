.VenuesContainer {
  width: 80%;
  margin: auto;
  min-width: 350px;
  height: 100vh;

  .search-bar-container {
    background: $primary-color;
    height: 200px;
    padding: 2%;
  }

  .venues-container {
    background-color: #E5E5E5;
    height: 100%;
    padding: 20px;
  }
}