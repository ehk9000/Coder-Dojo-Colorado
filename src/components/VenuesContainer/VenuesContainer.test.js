import React from 'react';
import { shallow } from 'enzyme';
import { VenuesContainer, mapStateToProps } from './VenuesContainer';

describe('VenuesContainer', () => {
  let wrapper;
  const isLoading = false;
  const error = '';
  let mockVenues = [
    {
      id: 1,
      name: 'Venue One',
      notes: 'Here be Dragons'
    }
  ];

  beforeEach(() => {
    wrapper = shallow(
      <VenuesContainer 
        isloading={isLoading} 
        error={error}
        venues={mockVenues}
      />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot with error/loading', () => {
    wrapper.setProps({ error: 'Failed to fetch', loading: true });
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    const mockState = {
      isFetching: false,
      error: 'Failed to fetch'
    };
    const expected = {
      isLoading: false,
      error: 'Failed to fetch'
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
