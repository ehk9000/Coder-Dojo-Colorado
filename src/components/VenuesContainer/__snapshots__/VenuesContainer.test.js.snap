// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VenuesContainer should match snapshot 1`] = `
<section
  className="VenuesContainer"
>
  <article
    className="search-bar-container"
  >
    <Connect(VenueSearchBar)
      location="venues"
    />
  </article>
  <article
    className="venues-container"
  >
    <h3>
      Found 1 dojos in DENVER
    </h3>
    <VenueCard
      venue={
        Object {
          "id": 1,
          "name": "Venue One",
          "notes": "Here be Dragons",
        }
      }
    />
  </article>
</section>
`;

exports[`VenuesContainer should match snapshot with error/loading 1`] = `
<section
  className="VenuesContainer"
>
  <article
    className="search-bar-container"
  >
    <Connect(VenueSearchBar)
      location="venues"
    />
  </article>
  <article
    className="venues-container"
  >
    <h3>
      Found 1 dojos in DENVER
    </h3>
    <h4>
      Failed to fetch
    </h4>
    <VenueCard
      venue={
        Object {
          "id": 1,
          "name": "Venue One",
          "notes": "Here be Dragons",
        }
      }
    />
  </article>
</section>
`;
